test/externalTests/solc-js/asm-json.js

```diff
@@ -0,0 +1,45 @@
+const tape = require('tape');
+const fs = require('fs');
+const solc = require('../index.js');
+
+tape('ASM Json output consistency', function (t) {
+    t.test('Nested structure', function(st) {
+        const testdir = 'test/';
+        const output = JSON.parse(solc.compile(JSON.stringify({
+            language: 'Solidity',
+            settings: {
+                viaIR: true,
+                outputSelection: {
+                    '*': {
+                        '*': ['evm.legacyAssembly']
+                    }
+                }
+            },
+            sources: {
+                C: {
+                    content: fs.readFileSync(testdir + 'code_access_runtime.sol', 'utf8')
+                }
+            }
+        })));
+        st.ok(output);
+
+        function containsTarget(obj, target) {
```

Comment:
```suggestion
        function containsAssemblyItem(assemblyJSON, item) {
```

Also, I think it would be best to limit the search to the `legacyAssembly` subobject. It seems that now you're searching the whole object and there could be random stuff there. It would also make this check easier to understand - it took me a moment to understand what's happening here, because this is not at all what I expected.

Reply:
Also as a general remark for the future - we should at some point create a CI job that runs the emscripten binary with those tests that provide Standard JSON input/output. It would make this whole JS mess unnecessary...

Reply:
> Also, I think it would be best to limit the search to the `legacyAssembly` subobject. It seems that now you're searching the whole object and there could be random stuff there. It would also make this check easier to understand - it took me a moment to understand what's happening here, because this is not at all what I expected.

Fair enough, what did you expect then? In my mind it was fairly straightforward to search the result json for some target object and it also seemed reasonably unlikely to me to find the target in our specific case anywhere else in the output. It is not big deal in any case, I am changing it to only consider the `legacyAssembly` subobject.

> Also as a general remark for the future - we should at some point create a CI job that runs the emscripten binary with those tests that provide Standard JSON input/output. It would make this whole JS mess unnecessary...

Yes please!!